### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by bohaz**
> ## In this pull request

- [ ] Add Redux Toolkit (`npm install react-redux @reduxjs/toolkit`).
- [ ] Create a directory that will contain all your Redux logic (`/src/redux`)
- [ ] Configure a Redux store (`/src/redux/store.js)`

Define a `slice` of state for ` books` that:

- Includes an array of books (initial state: empty array)
- Includes a reducer that adds a book
- Includes a reducer that removes a book

- [ ] Define a `slice` of state for `categories ` that:

- Includes an array of categories (initial state: empty array)
- Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

- [ ] Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:

```
/src
    |---/components
    |---/redux
        |--- /books
                | booksSlice.js
        |--- /categories
                | categoriesSlice.js
        | store.js
    | index.js
```
## Happy review

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  categories: [],
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+  },
+});
```
Comment (V-Blaze) [inline src/redux/categories/categoriesSlice.js]: >Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

- You have done great so far :+1: You set up redux correctly, but there are some requirements you missed that need to be implemented before moving to the next milestone. :+1: :100: 
- [ ] Looking at the project requirement above, you can see that the category needs to have a reducer that checks status and return a string of `Under construction`. currently, you do not have any reducer implemented, I'll suggest you add a new reducer `checkStatus` and them implement the functionality. :+1: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @bohaz , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Clean and well-structured code✔
-  Descriptive pull request summary👋 
-  Meaningful pull request title✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (bohaz) [reply src/redux/categories/categoriesSlice.js]: Thanks, I appreciate your professional review.

**Review by usorfaitheloho — APPROVED** [review]
Hi @bohaz 👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**